src/modal/internal.tsx

```diff
@@ -96,12 +98,35 @@ function PortaledModal({
 
   // enable / disable body scroll
   useEffect(() => {
+    /**
+     * Resets performanceMetricLogged
+     * PerformanceMetricLogged false and componentLoadingCount 0,
+     * indicates that there were no components loading in the modal and it was loaded instantly.
+     * In that case emit 0 as timeToContentReadyInModal
+     */
+    const resetModalPerformanceData = () => {
+      if (loadStartTime.current !== 0 && componentLoadingCount.current === 0 && !performanceMetricLogged.current) {
+        PerformanceMetrics.modalPerformanceData({
```

Comment:
The function name is a bit misleading, since it does not only reset data, it actually _emits_ data. I would recommend to keep only the resetting-related part in this function.

Reply:
updated